c7n/resources/vpc.py

```diff
@@ -2728,45 +2728,53 @@ class CrossAZRouteTable(Filter):
                     - type: cross-az-nat-gateway-route
                 actions:
                   - notify
+
     """
     schema = type_schema('cross-az-nat-gateway-route')
     permissions = ("ec2:DescribeRouteTables", "ec2:DescribeNatGateways", "ec2:DescribeSubnets")
 
+    def resolve_subnets(self, resource, subnets):
+        subnet_ids = set()
+        for association in resource['Associations']:
+            if association.get('Main'):
+                subnet_ids.update({
+                    s['SubnetId'] for s in subnets if s['VpcId'] == resource['VpcId']})
```

Comment:
@kapilt does this actually do what we want?  If we're dealing with the main route table, then this code will return ALL subnets (in the same VPC) as associated subnets. I don't think that's what we want. For main route table, we should only return subnets (in the same VPC) that are not explicitly associated in other route tables right?